src/SQLQueryStress/DatabaseSelect.cs

```diff
@@ -84,75 +84,146 @@ public DatabaseSelect(Form1.QueryStressSettings settings)
+        private void Db_comboBox_Leave(object sender, EventArgs e)
         {
-            SaveLocalSettings();
-
-            var selectedItem = (string) db_comboBox.SelectedItem;
+            ComboBox cbSender = ((ComboBox)sender);
+            if ((cbSender.SelectedValue == null || cbSender.Text != cbSender.SelectedItem.ToString()) && cbSender.Items.Contains(cbSender.Text))
+            {
+                cbSender.SelectedItem = cbSender.Text;
+            }
+        }
 
-            var sql = "" + "SELECT name " + "FROM master..sysdatabases " + "ORDER BY name";
+        private void Db_comboBox_Enter(object sender, EventArgs e)
+        {
+            ComboBox cbSender = ((ComboBox)sender);
+            string _prevSelectedValue = cbSender.SelectedValue != null ? cbSender.SelectedValue.ToString() : string.Empty;
+            ReloadDatabaseList(sender);
 
-            using (var conn = new SqlConnection(_localMainConnectionInfo.ConnectionString))
+            if (cbSender.Items.Contains(_prevSelectedValue))
             {
-                var comm = new SqlCommand(sql, conn);
+                cbSender.SelectedItem = _prevSelectedValue;
+            }
+        }
 
-                var databases = new List<string>();
+        private void ReloadDatabaseList(object objDatabaseParam)
+        {
+            ComboBox dbComboboxParam = (ComboBox)objDatabaseParam;
+            string selectedComboBoxItem = (string)dbComboboxParam.SelectedItem;
+            string connectionString;
+            if (dbComboboxParam == db_comboBox)
+            {
+                SaveLocalSettings();
+                connectionString = _localMainConnectionInfo.ConnectionString;
+            }
+            else
+            {
+                pm_saveLocalSettings();
+                connectionString = _localParamConnectionInfo.ConnectionString;
+            }
+            
+                var sql = "" + "SELECT name " + "FROM master..sysdatabases " + "ORDER BY name";
 
-                try
+                using (var conn = new SqlConnection(connectionString))
                 {
-                    conn.Open();
+                    var comm = new SqlCommand(sql, conn);
 
-                    var reader = comm.ExecuteReader();
+                    var databases = new List<string>();
 
-                    while (reader.Read())
+                    try
                     {
-                        databases.Add((string) reader[0]);
+                        conn.Open();
+
+                        var reader = comm.ExecuteReader();
+
+                        while (reader.Read())
+                        {
+                            databases.Add((string)reader[0]);
+                        }
                     }
-                }
-                catch (SqlException ex)
-                {
+                    catch (SqlException ex)
+                    {
                     if (ex.Number != 4060)

```

Comment:
The two database drop-downs (under "Main Load Settings" and "Parameterization Settings") are being reloaded using very similar procedures, and I saw a slight improvement in code maintainability by merging the two procedures.  I will separate this into two separate procedures if requested.